tests/validation/tests/v3_api/test_cli.py

```diff
@@ -0,0 +1,344 @@
+                                           version="1.5.0", timeout=120)
+    remove_cli_resource("apps", initial_app["id"])
+    assert initial_app["state"] == "active"
+    assert initial_app["version"] == "1.5.0"
+
+
+def test_cli_app_upgrade(remove_cli_resource, rancher_cli: RancherCli):
+    rancher_cli.log.info("Testing Rolling Back Apps")
+    initial_app = rancher_cli.apps.install("openebs", "openebs",
+                                           version="1.5.0", timeout=120)
+    remove_cli_resource("apps", initial_app["id"])
+    assert initial_app["version"] == "1.5.0"
+    upgraded_app = rancher_cli.apps.upgrade(initial_app, version="1.6.0")
+    assert upgraded_app["state"] == "active"
+    assert upgraded_app["version"] == "1.6.0"
+
+
+def test_cli_app_rollback(remove_cli_resource, rancher_cli: RancherCli):
+    rancher_cli.log.info("Testing Deleting Apps")
+    initial_app = rancher_cli.apps.install("openebs", "openebs",
+                                           version="1.5.0", timeout=120)
+    remove_cli_resource("apps", initial_app["id"])
+    assert initial_app["version"] == "1.5.0"
+    upgraded_app = rancher_cli.apps.upgrade(initial_app, version="1.6.0")
+    assert upgraded_app["version"] == "1.6.0"
+    rolled_back_app = rancher_cli.apps.rollback(upgraded_app, "1.5.0")
+    assert rolled_back_app["state"] == "active"
+    assert rolled_back_app["version"] == "1.5.0"
+
+
+def test_cli_app_delete(rancher_cli: RancherCli):
+    rancher_cli.log.info("Testing Deleting Apps")
+    initial_app = rancher_cli.apps.install("openebs", "openebs",
+                                           version="1.5.0", timeout=120)
+    deleted = rancher_cli.apps.delete(initial_app)
+    assert deleted
+
+
+@if_test_multicluster
+def test_cli_multiclusterapp_install(custom_cluster, remove_cli_resource,
+                                     rancher_cli: RancherCli):
+    rancher_cli.log.info("Testing Installing Multi-Cluster Apps")
+    # Get list of projects to use and ensure that it is 2 or greater
+    client = get_admin_client()
+    projects = client.list_project()
+    targets = []
+    for project in projects:
+        if project["name"] == "Default":
+            rancher_cli.switch_context(project['id'])
+            cluster_name, project_name = rancher_cli.get_context()
+            if cluster_name in [custom_cluster.name, CLUSTER_NAME]:
+                rancher_cli.log.debug("Using cluster: %s", cluster_name)
+                targets.append(project["id"])
+    assert len(targets) > 1
+
+    initial_app = rancher_cli.mcapps.install("openebs", targets=targets,
+                                             role="cluster-owner",
+                                             values=MULTICLUSTER_APP_ANSWERS,
+                                             version="1.5.0", timeout=120)
+    remove_cli_resource("mcapps", initial_app["name"])
+    assert initial_app["state"] == "active"
```

Comment:
We should check the targets at the app level to see if they have the right version/active/have been deleted